### PR TITLE
Movie tab rv feature

### PR DESCRIPTION
**Pull request by Obehi234**
Completed Movie Tab UI and functionality in Screen 2.

moviesTab

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -99,6 +99,7 @@ dependencies {
     implementation "com.google.android.material:material:$viewpager_version"
 
 
-
+    //Glide
+    implementation "com.github.bumptech.glide:glide:4.16.0"
```
Comment (Ruthy4) [inline app/build.gradle:103]: remove if no longer in use

File: app/src/main/java/com/example/streammoviesapplication/data/model/localData/MovieResultEntity.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    val popularity: Double? = 0.0,
+    val poster_path: String? = "",
+    val release_date: String? = "",
+    val title: String? = "",
+    val video: Boolean? = false,
+    val vote_average: Double? = 0.0,
+    val vote_count: Int? = 0
+)
+
+val result = listOf(
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/model/localData/MovieResultEntity.kt:26]: remove this

File: app/src/main/java/com/example/streammoviesapplication/data/repository/MoviesTabRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    private val api: MovieService,
+    private val movieTabDao: MovieListDao
+): IMoviesTabRepository {
+    override suspend fun fetchMoviesListTab(): Flow<Resource<List<MovieResultEntity>>> {
+        return flow {
+            when(val response = safeApiCall {  api.getMovieList()}) {
+                is Resource.Success -> {
+                    val movieTabResult = response.data?.results
+                    val movieTabList = movieTabResult?.map { result ->
+                        Log.d("CHECK_REPOSITORY", "$movieTabResult")
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/repository/MoviesTabRepositoryImpl.kt:25]: remove logs

File: app/src/main/java/com/example/streammoviesapplication/data/repository/MoviesTabRepositoryImpl.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                    emit(Resource.Error("${response.message}"))
+
+                }
+
+                is Resource.Loading -> {
+                    emit(Resource.Loading())
+                    Log.d("CHECK_LOADING", "Data Loading, please wait...")
+                }
+                else ->{
+                    Log.d("CHECK_ELSE", "I'M INSIDE THE ELSE BLOCK")
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/repository/MoviesTabRepositoryImpl.kt:48]: remove logs

File: app/src/main/java/com/example/streammoviesapplication/presentation/adapter/MovieListAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieListViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding = MovieTabRecyclerItemBinding.inflate(inflater, parent, false)
+        return MovieListViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: MovieListViewHolder, position: Int) {
+        Log.d("CHECK_VIEWHOLDER", "I'M IN VIEWHOLDER")
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/adapter/MovieListAdapter.kt:46]: remove logs

File: app/src/main/java/com/example/streammoviesapplication/presentation/adapter/MovieViewPagerAdapter.kt [hunk 6]
```diff
@@ -5,11 +5,12 @@ import android.view.ViewGroup
...
+import com.example.streammoviesapplication.databinding.HomeNavMoviePosterItemBinding
+
 
 class MovieViewPagerAdapter(private val imageList: List<TrendingMoviesEntity>) : RecyclerView.Adapter<MovieViewPagerAdapter.ViewPagerHolder>(){
 
-    inner class ViewPagerHolder(val binding: MoviePosterItemBinding): RecyclerView.ViewHolder(binding.root) {
+    inner class ViewPagerHolder(val binding: HomeNavMoviePosterItemBinding): RecyclerView.ViewHolder(binding.root) {
         fun bind(item: TrendingMoviesEntity) {
             binding.apply{
                 ivPoster.load("https://image.tmdb.org/t/p/w500" + item.poster_path)
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/adapter/MovieViewPagerAdapter.kt:16]: extract URL to constants

File: app/src/main/java/com/example/streammoviesapplication/presentation/adapter/TrendingViewPager.kt [hunk 7]
```diff
@@ -6,12 +6,13 @@ import androidx.recyclerview.widget.RecyclerView
...
+
 
 class TrendingViewPager (private val imageList : List<TrendingMoviesEntity>, private val viewPager2: ViewPager2)
     : RecyclerView.Adapter<TrendingViewPager.CardViewHolder>(){
 
-    class CardViewHolder(val binding: MoviePosterItemBinding) : RecyclerView.ViewHolder(binding.root) {
+    class CardViewHolder(val binding: HomeNavMoviePosterItemBinding) : RecyclerView.ViewHolder(binding.root) {
         fun bind(item: TrendingMoviesEntity) {
             binding.apply{
                 ivPoster.load("https://image.tmdb.org/t/p/w500" + item.poster_path)
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/adapter/TrendingViewPager.kt:18]: extract URL

File: app/src/main/java/com/example/streammoviesapplication/presentation/tabAdapters/MovieListAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import com.example.streammoviesapplication.databinding.MovieTabRecyclerItemBinding
+
+
+class MovieListAdapter() : ListAdapter<MovieResultEntity, MovieListAdapter.MovieListViewHolder>(MovieListDiffUtilCallback()) {
+
+    inner class MovieListViewHolder(private val binding : MovieTabRecyclerItemBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: MovieResultEntity) {
+
+            binding.apply{
+                movieCardBg.load("https://image.tmdb.org/t/p/w500" + item.poster_path)
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/tabAdapters/MovieListAdapter.kt:19]: extract url

File: app/src/main/java/com/example/streammoviesapplication/presentation/tabViews/TabMovieListFragment.kt [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        setUpRV()
+
+    }
+
+
+    private fun setUpRV() {
+        vm.moviesTabLiveData.observe(viewLifecycleOwner) { resource ->
+            when (resource) {
+                is Resource.Success -> {
+                    Log.d("CHECK_RESULT", "${resource.data}")
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/tabViews/TabMovieListFragment.kt:43]: remove logs

File: app/src/main/java/com/example/streammoviesapplication/presentation/tabViews/TabMovieListFragment.kt [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+                    Toast.makeText(context, "${resource.message}", Toast.LENGTH_LONG).show()
+                    Log.d("CHECK_TOAST", "${resource.message}")
+                }
+
+                is Resource.Loading -> {
+                    showProgressbar()
+                }
+
+                else -> {
+                    Log.d("CHECK_ELSE", "I'M IN ELSE")
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/tabViews/TabMovieListFragment.kt:61]: remove logs

File: app/src/main/java/com/example/streammoviesapplication/presentation/viewmodel/tabViewModel/TabViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+
+    private val _moviesTabLiveData = MutableLiveData<Resource<List<MovieResultEntity>>>()
+    val moviesTabLiveData: LiveData<Resource<List<MovieResultEntity>>> = _moviesTabLiveData
+
+    init {
+        fetchTabMovies()
+    }
+     fun fetchTabMovies() {
+         Log.d("CHECK_TAB","")
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/presentation/viewmodel/tabViewModel/TabViewModel.kt:29]: remove logs

File: app/src/main/res/layout/movie_tab_recycler_item.xml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:layout_marginTop="@dimen/margin_20dp"
+        app:cardCornerRadius="@dimen/radius_15dp"
+        app:cardElevation="@dimen/elevation_4dp" >
+
+        <ImageView
+            android:id="@+id/movie_card_bg"
+            android:layout_width="154dp"
+            android:layout_height="184dp"
```
Comment (Ruthy4) [inline app/src/main/res/layout/movie_tab_recycler_item.xml:23]: extract dimensions

**Review by IfyUnoka4 — COMMENTED** [review]
Remove all unused logs. Format code properly

**Review by IfyUnoka4 — APPROVED** [review]

**Pull request merged**